strapi/strapi
# Allow setting description for component and media fields

### What does it do?

Added description support for the component and media field types. It will be shown below the field label to avoid situations when the description would be pushed down due to a large media preview or a long list of repeatable components.

### Why is it needed?

It would be great to have an ability to set a description for any field to improve authors' & editors' experience.

### How to test it?

Go to Content-Types Builder, select a content type with a component or media field, go to the view configuration, edit options of the component or media field, set a description and save. The description should be visible in the content editor below the field label.

### Preview

![Screen Shot 2021-05-11 at 09 38 39](https://user-images.githubusercontent.com/920747/117777213-acb11380-b23c-11eb-9188-b9cfff917cc6.png)
![Screen Shot 2021-05-11 at 09 38 58](https://user-images.githubusercontent.com/920747/117777260-b89cd580-b23c-11eb-9091-d95daa8f6c28.png)


### TODO

- [ ] Green light from the core team
- [ ] Maybe move out description component styles to a separate component?

What do you think about it? Would you be willing to merge it once I have proper tests?
